server/pokemon-showdown/Dockerfile

```
@@ -0,0 +1,98 @@
+FROM node:24-bookworm-slim@sha256:3638d9a6fe4030bd716be989438248074489337ba3275657f93595428be4fc03 AS base
+
+# Every stage uses the same absolute path so artifacts can move between stages
+# without path translation.
+WORKDIR /opt/pokemon-showdown
+
+FROM base AS build
+
+# Git retrieves the pinned source, and CA certificates verify GitHub's HTTPS
+# connection. Package metadata is removed in the same layer to keep it small.
+# droast ignore=DF005 reason="the base image digest already fixes the available package set; a Debian version pin breaks the build at the next point release, when the archive drops the old version"
+RUN apt-get update \
```

Hadolint (2.15.1)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)